test/extractors/CSVStagingExtractor.test.js

```diff
@@ -0,0 +1,70 @@
+const path = require('path');
+const rewire = require('rewire');
+const _ = require('lodash');
+const { CSVStagingExtractor } = require('../../src/extractors');
+const exampleStagingModuleResponse = require('./fixtures/csv-staging-module-response.json');
+const exampleCSVStagingBundle = require('./fixtures/csv-staging-bundle.json');
+
+// Constants for tests
+const MOCK_PATIENT_MRN = 'pat-mrn-1'; // linked to values in example-module-response above
+const MOCK_CSV_PATH = path.join(__dirname, 'fixtures/example.csv'); // need a valid path/csv here to avoid parse error
+
+// Instantiate module with parameters
+const csvStagingExtractor = new CSVStagingExtractor({ filePath: MOCK_CSV_PATH });
+
+// Destructure all modules
+const { csvModule } = csvStagingExtractor;
+
+// Spy on csvModule
+const csvModuleSpy = jest.spyOn(csvModule, 'get');
+
+const formatTNMCategoryData = rewire('../../src/extractors/CSVStagingExtractor.js').__get__('formatTNMCategoryData');
+
+describe('CSVStagingExtractor', () => {
+  describe('formatTNMCategoryData', () => {
+    test('should join data appropriately and throw errors when missing required properties', () => {
+      const expectedErrorString = 'Staging data is missing an expected property: mrn, conditionId, effectiveDate are required.';
+      const localData = _.cloneDeep(exampleStagingModuleResponse[0]);
+      // Test that valid data works fine
+      expect(formatTNMCategoryData(localData)).toEqual(expect.anything());
+
+      // Test all required properties are
```

Comment:
Could we elaborate on this a bit more? 
```suggestion
      // Test all optional properties can be removed without throwing errors
```

Reply:
ashamed to say this is straight copy and paste 😂 , will elaborate!